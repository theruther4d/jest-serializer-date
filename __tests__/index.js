test('Should handle current dates', () => {
    expect({date: new Date()}).toMatchSnapshot()
})

test('Should handle non-current dates', () => {
    expect({date: new Date(2018, 1, 1)}).toMatchSnapshot()
})

test('Should ignore non-date objects', () => {
    expect({notADate: {something: 'else'}}).toMatchSnapshot()
})

test('Should ignore non-objects', () => {
    let val = {
        string: 'string',
        bool: true,
        undef: undefined,
        obj: {not: 'an object'},
        num: 1,
        arr: ['nope', 'nope', 'nope']
    }
    expect(val).toMatchSnapshot()
})