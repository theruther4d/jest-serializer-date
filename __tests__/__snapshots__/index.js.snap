// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should handle current dates 1`] = `
Object {
  "date": Current Date,
}
`;

exports[`Should handle non-current dates 1`] = `
Object {
  "date": 1-1-2018,
}
`;

exports[`Should ignore non-date objects 1`] = `
Object {
  "notADate": Object {
    "something": "else",
  },
}
`;

exports[`Should ignore non-objects 1`] = `
Object {
  "arr": Array [
    "nope",
    "nope",
    "nope",
  ],
  "bool": true,
  "nan": NaN,
  "null": null,
  "num": 1,
  "obj": Object {
    "not": "an object",
  },
  "string": "string",
  "undef": undefined,
}
`;
